.github/actions/bundle_size_test/action.yml

```diff
@@ -1,8 +1,34 @@
-name: bundlesize
+name: bundle_size_test
+inputs:
+  cwd:
+    required: true
+  config:
+    required: true
 runs:
   using: composite
   steps:
-  - name: Check bundle size
+  - name: Analyze build
+    working-directory: ${{ inputs.cwd }}
     run: |-
-        npm run test:max-file-size
+        npm run analyze-build
+    shell: bash
+
+  - name: Check bundle sizes
+    working-directory: ${{ inputs.cwd }}
+    # NOTE: A bug prevented us from running bundlesize with a config file:
+    # npx bundlesize --scope ${{ github.action_path }}/bundlesize.config.json
+    # For more details, see https://github.com/siddharthkp/bundlesize/issues/326
+    run: |-
+        json_string='${{ inputs.config }}'
+
+        # Parse the given json string and build the command to run
+        echo $json_string | jq -r 'to_entries[] | "npx bundlesize -f \"" + .key + "\" -s " + .value' | while read -r command; do
```

Comment:
Should we add some error handling in the jq parsing and print a message to the command failed?